eogrow/utils/validators.py

```diff
@@ -51,34 +51,44 @@ def optional_validator(value, values, config, field):  # type: ignore[no-untyped
     return validator(field, allow_reuse=allow_reuse, **kwargs)(optional_validator)
 
 
-def ensure_exactly_one_defined(param1: str, param2: str, allow_reuse: bool = True, **kwargs: Any) -> classmethod:
-    """A root validator that makes sure only one of the two parameters is defined."""
+def ensure_exactly_one_defined(first_param: str, second_param: str, **kwargs: Any) -> classmethod:
+    """A validator (applied to `second_param` field) that makes sure only one of the two parameters is defined.
 
-    def ensure_exclusion(cls: type, values: RawSchemaDict) -> RawSchemaDict:
-        is_param1_defined = values.get(param1) is None
-        is_param2_defined = values.get(param2) is None
+    Make sure that the definition of `second_param` comes after `first_param˙ (in line-order).
+    """
+
+    def ensure_exclusion(cls: type, value: Optional[Any], values: RawSchemaDict) -> Optional[Any]:
+        is_param1_defined = values.get(first_param) is None
+        is_param2_defined = value is None
```

Comment:
Doesn't change anything down the line I think, but these should probably be negated, or the name changed

Reply:
adjusted names, it is indeed wrong to have it like it is. Probably a mistake due to simplifications